src/etos_test_runner/lib/iut_monitoring.py

```diff
@@ -14,25 +14,79 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """IUT monitoring module."""
+import sys
+import os
+import stat
+import logging
+from threading import Thread
+from signal import SIGINT
+from subprocess import Popen, PIPE, STDOUT, TimeoutExpired
 from etos_lib.lib.config import Config
 
-# TODO: Get this from environment provider
+
+ON_POSIX = 'posix' in sys.builtin_module_names
 
 
 class IutMonitoring:
     """Helper class for monitoring IuT health statistics."""
 
+    logger = logging.getLogger("IUT Monitoring")
+
     def __init__(self, iut):
         """Initialize monitoring.
 
         :param iut: IUT object to monitor.
         :type iut: :obj:`etr.lib.iut.Iut`
         """
         self.iut = iut
+        self.processes = []
         self.config = Config()
 
+    def _read_from_process(self, output):
+        """Non-blocking read from a process output.
+
+        :param output: Output to read from.
+        :type output: filedescriptor
+        """
+        self.logger.info("Reading output from %r", output)
+        for line in iter(output.readline, b''):
+            self.logger.info(line.decode("utf-8"))
+        output.close()
+
     def start_monitoring(self):
         """Start monitoring IUT."""
+        scripts = self.config.get("scripts") or []
```

Comment:
This assumes a proper shebang. Would it be wrong to only support for instance bash? or python? or is that to restrictive?

Reply:
I believe that it would be a bit too restrictive.
We cannot foresee what others want to do.